core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -267,10 +277,22 @@ object SparkSubmit {
     // In yarn-cluster mode, use yarn.Client as a wrapper around the user class
     if (isYarnCluster) {
       childMainClass = "org.apache.spark.deploy.yarn.Client"
-      if (args.primaryResource != SPARK_INTERNAL) {
-        childArgs += ("--jar", args.primaryResource)
+      if (args.isPython) {// yarn-cluster mode for python application
+      val primaryResourceLocalPath = new Path(args.primaryResource)
+        childArgs += ("--primaryResource", primaryResourceLocalPath.getName)
+        val pyFilesLocalNames:String = if (args.pyFiles != null) {
+          args.pyFiles.split(",").map { p => (new Path(p)).getName }.mkString(",")
+        } else {
+          "null"
+        }
+        childArgs += ("--py-files", pyFilesLocalNames.toString)
```

Comment:
No need for `toString`, this is already a string.  Also, can we avoid adding this arg at all instead of using "null"?